backend/session/direct-ipc.c

```diff
@@ -158,18 +162,24 @@ static void communicate(int sock) {
 				ret = errno;
 			}
 error:
+			if (ret == 0) {
```

Comment:
This function also handles opening input devices, so you're remembering and later trying to drmDropMaster them too.

Reply:
I misread the ENOTSUP if-statement above. I'll make the remembering conditional on DRM_MAJOR.